app/controllers/campaigns_controller.rb

```diff
@@ -43,6 +43,21 @@ def destroy
     end
   end
 
+  def publish
+    redirect_to campaigns_path if @campaign.publish!
```

Comment:
Can `@campaign.publish!` fail? What will happen if you are trying to publish already published campaign for example?